### PR TITLE
Updated the parser to be able to handle multiple formats for model

### PR DESCRIPTION
**Pull request by ivakegg**
and model field attributes.  Updated tests and documentation.

### CODE REVIEW

File: src/main/java/datawave/query/model/ModelKeyParser.java [hunk 1]
```diff
@@ -51,12 +84,25 @@ public static FieldMapping parseKey(Key key, Value value) {
         // we can have attributes no matter the mapping
         attributes.addAll(Arrays.asList(StringUtils.split(new String(value.get(), StandardCharsets.UTF_8), ',')));
         
-        if (1 == colq.length) {
-            // in this case we expect model field attributes (not specific to a mapping)
+        if (0 == colq.length || (colq.length == 1 && colq[0].isEmpty())) {
```
Comment (jschmidt10) [inline src/main/java/datawave/query/model/ModelKeyParser.java:87]: Your example keys above are awesome. Thanks! There's doesn't appear to be one that matches this section of the parsing though.

Comment (ivakegg) [reply src/main/java/datawave/query/model/ModelKeyParser.java:87]: oh right, I was also supporting that case at one point.  That is probably too much, I will make this one error out.

**Review by hlgp — APPROVED** [review]

File: src/main/java/datawave/query/model/ModelKeyParser.java [hunk 2]
```diff
@@ -11,7 +12,37 @@
...
+ *   "model" MODEL_NAME:ATTRIBUTE
+ *   "model" MODEL_NAME:ATTR_NAME ATTR_VALUES
+ *   "model" MODEL_NAME:"attrs" ATTRIBUTES
+ *   MODEL_FIELD MODEL_NAME:ATTRIBUTE
+ *   MODEL_FIELD MODEL_NAME:ATTR_NAME ATTR_VALUES
+ *   MODEL_FIELD MODEL_NAME:"attrs" ATTRIBUTES
+ *
+ * deprecated but parsable format:
+ *   MODEL_FIELD MODEL_NAME:DB_FIELD\x00"index_only\x00"forward" ATTRIBUTES
+ * @formatter:on
```
Comment (hlgp) [inline src/main/java/datawave/query/model/ModelKeyParser.java:44]: // @formatter:on

Comment (ivakegg) [reply src/main/java/datawave/query/model/ModelKeyParser.java:44]: oh, am I doing that wrong.  ok

Comment (ivakegg) [reply src/main/java/datawave/query/model/ModelKeyParser.java]: fixed

File: src/main/java/datawave/query/model/ModelKeyParser.java [hunk 3]
```diff
@@ -11,7 +12,37 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 
+/**
+ * This class can be used to parse keys belonging to a model.
+ *
+ * @formatter:off
```
[the remark this answers is not in the document]
Comment (ivakegg) [reply src/main/java/datawave/query/model/ModelKeyParser.java]: fixed

**Review by hlgp — APPROVED** [review]

**Review by jschmidt10 — APPROVED** [review]

File: src/main/java/datawave/query/model/ModelKeyParser.java [hunk 4]
```diff
@@ -51,12 +84,19 @@ public static FieldMapping parseKey(Key key, Value value) {
         // we can have attributes no matter the mapping
         attributes.addAll(Arrays.asList(StringUtils.split(new String(value.get(), StandardCharsets.UTF_8), ',')));
         
-        if (1 == colq.length) {
-            // in this case we expect model field attributes (not specific to a mapping)
+        if (0 == colq.length || (colq.length == 1 && colq[0].isEmpty())) {
```
Comment (d-hwang) [inline src/main/java/datawave/query/model/ModelKeyParser.java:87]: nitpick, but 0 == colq.length will never happen. also i'm surprised that ModelKeyParser.parseKey(VERSION_KEY1, VERSION_VALUE1) doesn't throw IAE where CQ is an empty string.

Comment (ivakegg) [reply src/main/java/datawave/query/model/ModelKeyParser.java:87]: OK, This is weird, but that test case actually fails in my IDE....   Not convinced the tests below actually did their job.... DOH!

**Pull request merged**